File: leetcode/top-interview-questions-easy/strings/4. valid_anagram.js
/**
 * @param {string} s
 * @param {string} t
 * @return {boolean}
 */
var isAnagram = function (s, t) {
  if (s.length !== t.length) {
    return false;
  }

  var map = {};
  for (var char of s) {
    const count = map[char] || 0;
    map[char] = count + 1;
  }

  for (var char of t) {
    const count = map[char] || 0;
    map[char] = count - 1;
  }

  for (var key of Object.values(map)) {
    if (key !== 0) {
      return false;
    }
  }

  return true;
};

File: leetcode/top-interview-questions-easy/strings/4. valid_anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s) != len(t):
            return False
        
        map = {}
        for char in s:
            count = map.get(char, 0)
            # Ну эта задача уже легче пошла переписываться
            # Но когда есть get но нет set - это оригинально
            map[char] = count + 1
            
        for char in t:
            count = map.get(char, 0)
            map[char] = count - 1
            
        for key in map.values():
            if key != 0:
                return False
            
        return True
        
       